lib/utils/utf8.c

```diff
@@ -79,3 +80,37 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const char *str, size_t maxlen)
+{
+	int i = 0, nbyte = 0;
+	size_t len = strlen(str);
+	unsigned char *buf = (unsigned char *)str;
+
+	/* Return false when the string is greater than the maximum length. */
+	if (len > maxlen) {
+		return false;
+	}
```

Comment:
This depends on the definition of `maxlen`. The one I suggested above `maxlen` refers to the maximum number of octets that is considered here, to help prevent accessing invalid data if the provided `str` is not NULL terminated. This is similar to how the `strnXXX` functions work. 

However the way it is used here, if `str = "abcabc"` and `maxlen = 3`, then this returns false, even if the first 3 octets are a valid UTF-8 string. 

Is that intentional? The behavior implemented here is both atypical and undocumented

Reply:
Here I assume that the user knows the maximum length of the string (actually, it is only because strnlen cannot be used, because it cannot pass ci, it is a function of posix, not stdc

Reply:
> it is only because strnlen cannot be used, because it cannot pass ci, it is a function of posix, not stdc

I don't follow. Why can't `strnlen` be used? 

Either way, the semantics of `maxlen` is still unclear IMO

Reply:
My initial change was to use strnlen, but CI gave me an error with the syntax. However, in order to implement the function safely, I had to add maxlen.

Reply:
Can you please share the issues? 
There really should not be any issues with using `const char *`, and if there is, we have much bigger issuers in Zephyr than this function :D